browser-extension/mv2/src/background/background.js

```diff
@@ -1067,6 +1071,18 @@ BG.Methods.onContentScriptLoadedNotification = async (tabId) => {
       () => window.tabService.removeData(tabId, "appliedRuleDetails")
     );
   }
+
+  if (window.tabService.getData(tabId, "recordTab") === true) {
+    chrome.tabs.sendMessage(tabId, { action: RQ.CLIENT_MESSAGES.START_RECORDING }, () =>
+      window.tabService.removeData(tabId, "recordTab")
+    );
+  }
+};
+
+BG.Methods.startRecordingOnUrl = (url) => {
+  chrome.tabs.create({ url }, (tab) => {
+    window.tabService.setData(tab.id, "recordTab", true);
```

Comment:
`setData` works only when an entry for the tab already exists in the map of tabs and the entry won't be there if `chrome.tabs.onUpdated` is not called immediately on `tabs.create` in some browser.
```js
setData: function (tabId, key, value) {
      if (!this.map.hasOwnProperty(tabId)) { // <-----
        return;
      }
     ...
}
```

Reply:
Please ignore, as the map entry is added on `onCreated` event and not on `onUpdated` event.